service/interpreter/activityImpl.go

```diff
@@ -180,16 +180,29 @@ func composeHttpError(
 		maxL := len(errMsg)
 		if maxL > 5 {
 			maxL = 5
+			errMsg = errMsg[:maxL] + "..."
 		}
-		errMsg = errMsg[:maxL] + "..."
 
 		maxL = len(responseBody)
 		if maxL > 50 {
 			maxL = 50
+			responseBody = responseBody[:maxL] + "..."
 		}
-		responseBody = responseBody[:maxL] + "..."
 		errType = "1st-attempt-failure"
+	} else {
```

Comment:
Nit: maybe nice to create a small func for the repeated code

Reply:
Refactored

Reply:
Looks good